src/App/Pages/Settings/SettingsPage/SettingsPageViewModel.cs

```diff
@@ -326,6 +329,45 @@ public async Task LoggerReportingAsync()
             BuildList();
         }
 
+        public async Task ApproveLoginRequestsAsync()
+        {
+            var options = new[]
+            {
+                    CreateSelectableOption(AppResources.Yes, _approvePasswordlessLoginRequests),
+                    CreateSelectableOption(AppResources.No, !_approvePasswordlessLoginRequests),
+            };
+
+            var selection = await Page.DisplayActionSheet(AppResources.UseThisDeviceToApproveLoginRequests, AppResources.Cancel, null, options);
+
+            if (selection == null || selection == AppResources.Cancel)
+            {
+                return;
+            }
+
+            _approvePasswordlessLoginRequests = CompareSelection(selection, AppResources.Yes);
+            await _stateService.SetApprovePasswordlessLoginsAsync(_approvePasswordlessLoginRequests);
+
+            BuildList();
+
+            if (!_approvePasswordlessLoginRequests)
+            {
+                return;
+            }
+
+            var notificationsSettingsEnabled = await _pushNotificationService.GetNotificationsSettingsEnabledAsync();
+            if (notificationsSettingsEnabled)
+            {
+                return;
+            }
+
+            var openAppSettingsResult = await _platformUtilsService.ShowDialogAsync(AppResources.ReceivePushNotificationsForNewLoginRequests, title: string.Empty, confirmText: AppResources.Settings, cancelText: AppResources.NoThanks);
+            if (openAppSettingsResult)
+            {
+                _deviceActionService.OpenAppSettings();
+            }
```

Comment:
you can put everything in one if:
```c#
if (_approvePasswordlessLoginRequests
    &&
    !await _pushNotificationService.GetNotificationsSettingsEnabledAsync()
    &&
    await _platformUtilsService.ShowDialogAsync(AppResources.ReceivePushNotificationsForNewLoginRequests, title: string.Empty, confirmText: AppResources.Settings, cancelText: AppResources.NoThanks))
{
    _deviceActionService.OpenAppSettings();
}
```
or at least the first two if you think the third condition is two lengthy to put it on the if:
```c#
if (!_approvePasswordlessLoginRequests
    ||
    await _pushNotificationService.GetNotificationsSettingsEnabledAsync())
{
    return;
}
```

Reply:
i'll change for the second option. I usually don't like to include method lengthy calls into if statements but i guess it cleans up a couple of lines in this case.